pkg/v5/configmap/spec.go

```diff
@@ -0,0 +1,19 @@
+package configmap
+
+import (
+	"context"
+
+	corev1 "k8s.io/api/core/v1"
+)
+
+type Interface interface {
+	GetDesiredState(ctx context.Context, configMapValues ConfigMapValues) ([]*corev1.ConfigMap, error)
+}
+
+// ConfigMapValues is used by the configmap resources to provide data to the
+// configmap service.
+type ConfigMapValues struct {
+	ClusterID    string
+	Organization string
+	WorkerCount  int
+}
```

Comment:
This struct is used by the resource to provide data. For managed services the idea is to keep the values to as few as possible. We should use good defaults in the Helm charts whenever possible.

If the number of values increases we can structure this more.

Reply:
I agree that this holds true here. I looked at this with the perspective of host-cluster and have some issues in my mind dealing with this. 
Would we need a struct of the entire installations repo for host cluster to handle this?

Reply:
Yes for managed services the number of values is low. We also want to provide the cloud provider to the Ingress Controller chart so we can create the load balancer service. That's all I'm aware of currently.

For host cluster and release-operator there is more config and I think we'll need to structure more. What I like about the current design is we just create the chartconfig CR and a corresponding config map and secret. How we create those is up to us.

For Managed Services and Service Catalog we'll probably need to restructure this. As we'll be managing a lot more charts and some of them will have config maps. I'm thinking we may need to split out to a catalog-operator to prevent cluster-operator from becoming huge.

This current approach will let us complete the k8scloudconfig migration. 

@MarcelMue I hope this helps but it might not be the answer you wanted?

cc @puja108 @fgimenez

Reply:
Not the answer I hoped for, but the one I expected :D 
All good, I'll try to get my head around a solution.